File: 백준/Silver/15649. N과 M （1）/N과 M （1）.cc
#include <iostream>
using namespace std;

int n, m;
int a[10];
bool isused[10];

void func(int k) {
    if(k == m) {
       for(int i = 0; i < m; i++) {
           cout << a[i] << ' ';
       }
        cout << '\n';
       return;
    }
    for(int i = 1; i <= n; i++ ){
        if(!isused[i]) {
            a[k] = i;
            isused[i] = 1;
            func(k+1);
            isused[i] = 0;
        }
    }
        
}

int main() {
    cin >> n >> m;
    func(0);
    
}

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().trim().split('\n');

const[n,m] = input[0].split(' ').map(Number);

let a = [];
let isused = new Array(10).fill(0);

function func(k) {
    if(k === m) {
        console.log(a.join(' '));
        return;
    }

    for(let i = 1; i<= n; i++) {
        if(!isused[i]) {
            a[k] = i;
            isused[i] = 1;
            func(k+1);
            isused[i] = 0;
        }
    }
}

func(0);


